src/plugins/teleop/Teleop.cc

```diff
@@ -0,0 +1,274 @@
+  ignmsg << "[OnlinearVelSelection]: linear velocity: "
+      << this->dataPtr->linearVel << std::endl;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnAngularVelSelection(const QString& _velocity)
+{
+  this->dataPtr->angularVel = _velocity.toDouble();
+  ignmsg << "[OnlinearVelSelection]: angular velocity: "
+      << this->dataPtr->angularVel << std::endl;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnKeySwitch(bool _checked)
+{
+  this->dataPtr->linearDir = 0;
+  this->dataPtr->angularDir = 0;
+  this->dataPtr->keyEnable = _checked;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnSlidersSwitch(bool _checked)
+{
+  if(_checked){
+    this->dataPtr->linearDir = 1;
+    this->dataPtr->angularDir = 1;
+    OnTeleopTwist();
+  }
+}
+
+/////////////////////////////////////////////////
+bool Teleop::eventFilter(QObject *_obj, QEvent *_event)
+{
+  if(_event->type() == QEvent::KeyPress && this->dataPtr->keyEnable == true)
+  {
+    QKeyEvent *keyEvent = static_cast<QKeyEvent*>(_event);
+    switch(keyEvent->key())
+    {
+      case Qt::Key_W:
+        this->dataPtr->linearState = keyLinear::forward;
+        break;
+      case Qt::Key_A:
+        this->dataPtr->angularState = keyAngular::left;
+        break;
+      case Qt::Key_D:
+        this->dataPtr->angularState = keyAngular::right;
+        break;
+      case Qt::Key_X:
+        this->dataPtr->linearState = keyLinear::backward;
+        break;
+      default:
+        ignmsg << "A non valid key was pressed" << std::endl;
+        break;
+    }
+    setKeyDirection();
+    OnTeleopTwist();
+  }
+  if(_event->type() == QEvent::KeyRelease && this->dataPtr->keyEnable == true)
+  {
+    QKeyEvent *keyEvent1 = static_cast<QKeyEvent*>(_event);
+    switch(keyEvent1->key())
+    {
+      case Qt::Key_W:
+        this->dataPtr->linearState = keyLinear::stop;
+        break;
+      case Qt::Key_A:
+        this->dataPtr->angularState = keyAngular::stop;
+        break;
+      case Qt::Key_D:
+        this->dataPtr->angularState = keyAngular::stop;
+        break;
+      case Qt::Key_X:
+        this->dataPtr->linearState = keyLinear::stop;
+        break;
+      default:
+        ignmsg << "A non valid key was pressed" << std::endl;
+        break;
+    }
+    setKeyDirection();
+    OnTeleopTwist();
+  }
+  return QObject::eventFilter(_obj, _event);
+}
+
+/////////////////////////////////////////////////
+void Teleop::setKeyDirection()
+{
+  if(this->dataPtr->linearState == keyLinear::forward)
+    this->dataPtr->linearDir = 1;
+  else if(this->dataPtr->linearState == keyLinear::backward)
+    this->dataPtr->linearDir = -1;
+  else
+    this->dataPtr->linearDir = 0;
+
+  if(this->dataPtr->angularState == keyAngular::left)
+    this->dataPtr->angularDir = 1;
+  else if(this->dataPtr->angularState == keyAngular::right)
+    this->dataPtr->angularDir = -1;
+  else
+    this->dataPtr->angularDir = 0;
```

Comment:
This could be collapsed to:

```suggestion
this->dataPtr->linearDir = this->dataPtr->linearState == keyLinear::forward ? 1 : this->dataPtr->linearState == keyLinear::backward ? -1 : 0;
this->dataPtr->angularDir = this->dataPtr->angularState == keyAngular::left ? 1 : this->dataPtr->angularState == keyAngular::right ? -1 : 0;
```

Reply:
Done